glider_dac/glider_emails.py

```diff
@@ -62,3 +81,104 @@ def get_erddap_catalog_url():
+                          "completed": completed}
+
+            if not force:
+                q_dict["compliance_check_passed"] = {"$ne": True}
+
+        # TODO: combine username/deployment cases?
+        if username:
+            q_dict = {"username": username}
+        # a particular deployment has been specified
+        elif deployment_dir:
+            q_dict = {"deployment_dir": deployment_dir}
+        else:
+            q_dict = {}
+
+        agg_pipeline = [{"$match": q_dict},
+                        {"$group": {"_id": "$user_id",
+                           "deployments": {"$push":
+                                {"_id": "$_id",
+                                 "name": "$name",
+                                 "deployment_dir": "$deployment_dir"} } }
+                        }]
+        # if force is enabled, re-check the datasets no matter what
+
+        # is this syntax still used?  if the first fn call fails, use the
+        # second set of results
+        try:
+            agg_result_set = db.deployments.aggregate(agg_pipeline)['result']
+        except:
+            agg_result_set = db.deployments.aggregate(agg_pipeline, cursor={})
+        for res in agg_result_set:
+            user = db.users.find_one(res["_id"])
+            all_messages = []
+            failing_deployments = []
+            for dep in res['deployments']:
+                root_logger.info("Running compliance check on glider "
+                                 "deployment: {}".format(dep))
+                try:
+                    dep_passed, dep_messages = process_deployment(dep)
+                    all_messages.append(dep_messages)
+                    if not dep_passed:
+                        failing_deployments.append(dep)
+                except Exception as e:
+                    root_logger.exception(
+                        "Exception occurred while processing deployment {}".format(dep['name']))
+                    text_body = ''
+            send_deployment_cchecker_email(user, failing_deployments,
+                                           "\n".join(all_messages))
+
+def process_deployment(dep):
+    deployment_issues = "Deployment {}".format(os.path.basename(dep['name']))
+    groups = OrderedDict()
+    erddap_fmt_string = "erddap/tabledap/{}.nc?&time%3Emax(time)-1%20day"
+    base_url = app.config["PRIVATE_ERDDAP"]
+    # FIXME: determine a more robust way of getting scheme
+    if not base_url.startswith("http"):
+        base_url = "http://{}".format(base_url)
+    url_path = "/".join([base_url,
+                         erddap_fmt_string.format(dep["name"])])
+    # TODO: would be better if we didn't have to write to a temp file
+    outhandle, outfile = tempfile.mkstemp()
+    try:
+        failures, errors = ComplianceChecker.run_checker(ds_loc=url_path,
+                                      checker_names=['gliderdac'], verbose=True,
+                                      criteria='normal', output_format='text',
```

Comment:
Can we be lenient on the criteria please? We'll eventually have a thorough look at the priorities of the checks to ensure the high priority stuff is indeed high priority.

Reply:
Updated in 20f813c.